docs(k8s): review docs and tutorials 2024-11-04 (#3932)
scaleway/docs-content: containers/kubernetes/reference-content/multi-az-clusters.mdx

- Kapsule's Control Plane network access is managed by a Load Balancer in the primary zone of each region. If this zone fails globally, the Control Plane will be unreachable, even if the cluster spans multiple zones. This limitation also applies to HA Dedicated Control Planes.
- Persistent volumes are limited to their Availability Zone (AZ). Applications must replicate data across persistent volumes in different AZs to maintain high availability in case of zone failures.

> - In "controlled isolation" mode, nodes access the Control Plane via their public IPs. If two AZs can nott communicate (split-brain scenario), nodes will not appear unhealthy from Kubernetes' perspective, but communication between nodes in different AZs will be disrupted. Applications must handle this scenario if they use components across multiple AZs.

Reviewer: ```suggestion
- In "controlled isolation" mode, nodes access the Control Plane via their public IPs. If two AZs can not communicate (split-brain scenario), nodes will not appear unhealthy from Kubernetes' perspective, but communication between nodes in different AZs will be disrupted. Applications must handle this scenario if they use components across multiple AZs.
```

> - In "full isolation" mode, nodes also use the Public Gateway to access the Control Plane. If nodes can not reach the Public Gateway (e.g. because of Private Network failure in an AZ), they will become unhealthy. As there is only one Public Gateway per Private Network, losing the AZ with the Public Gateway results in the loss of all nodes in all private pools across all AZs.

Reviewer 2: ```suggestion
- In "full isolation" mode, nodes also use the Public Gateway to access the Control Plane. If nodes cannot reach the Public Gateway (e.g. because of Private Network failure in an AZ), they will become unhealthy. As there is only one Public Gateway per Private Network, losing the AZ with the Public Gateway results in the loss of all nodes in all private pools across all AZs.
```
